packages/frontend-plugin-api/src/extensions/createComponentExtension.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { lazy } from 'react';
+import {
+  AnyExtensionInputMap,
+  ExtensionInputValues,
+  coreExtensionData,
+  createExtension,
+} from '../wiring';
+import { Expand } from '../types';
+import { PortableSchema } from '../schema';
+import { ExtensionError, ExtensionBoundary } from '../components';
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import { AppComponents } from '../../../core-app-api/src/app/types';
+
+/** @public */
+export function createBootErrorPageExtension<
```

Comment:
The extensions here are very similar except for the input name and component type, I'd like to hear your suggestions on how to improve this. My initial idea would be to have a factory to `createCoreComponentExtension` that receives the input name. Any ideas?

Reply:
Using component ref instead.